packages/ai-chat/src/common/chat-model.ts

```diff
@@ -948,8 +964,18 @@ class ChatResponseImpl implements ChatResponse {
     }
 
     protected _updateResponseRepresentation(): void {
-        this._responseRepresentation = this._content
+        this._responseRepresentation = this.responseRepresentationsToString(this._content, false);
+        this._responseRepresentationForDisplay = this.responseRepresentationsToString(this.content, true);
+    }
+
+    protected responseRepresentationsToString(content: ChatResponseContent[], forDisplay: boolean): string {
```

Comment:
Nitpick: boolean parameters are always hard to parse and understand. Here it's easier as the caller and definition are side by side, even then one needs to mentally parse what happens in each call.

This could be refactored using a string, e.g.
```
  this._responseRepresentation = this.responseRepresentationsToString(this._content, 'asString');
  this._responseRepresentationForDisplay = this.responseRepresentationsToString(this.content, 'asDisplayString');

protected responseRepresentationsToString(content: ChatResponseContent[], collect: 'asString' | 'asDisplayString' {
```